### PR TITLE
Feature/textarea rebrand

### PR DESCRIPTION
**Pull request by fraincs**
- Applied the Hopper Brand to the Textarea component.

### CODE REVIEW

File: packages/Textarea/src/textarea.scss [hunk 1]
```diff
@@ -2,94 +2,145 @@
...
+        bottom: var(--ids-textarea-char-indicator-bottom);
+        color: var(--ids-textarea-char-indicator-color);
+        font-size: var(--ids-textarea-char-indicator-size);
+        line-height: var(--ids-textarea-char-indicator-line-height);
+        position: absolute;
+        right: var(--ids-textarea-char-indicator-right);
     }
+}
 
-    &--has-char-count {
```
Comment (vicky-comeau) [inline packages/Textarea/src/textarea.scss:82]: why was this removed?

File: packages/Textarea/src/textarea.scss [hunk 2]
```diff
@@ -2,94 +2,145 @@
...
-    color: var(--ids-textarea-char-indicator-color);
-    font-size: var(--ids-textarea-char-indicator-size);
-    line-height: var(--ids-textarea-char-indicator-line-height);
-    position: absolute;
-    right: var(--ids-textarea-char-indicator-right);
-  }
+[data-brand="workleap"] .ids-textarea__field:focus,
+[data-brand="workleap"] .ids-textarea--focus .ids-textarea__field {
+    box-shadow: var(--ids-textarea-focus-box-shadow);
 }
```
[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/Textarea/src/textarea.scss:146]: actually, igloo should have a focus style. I'm confused as to why you only specify workleap here

**Comment by vicky-comeau** [comment]
I don't think this is normal... 
image
It's on the docs so the file is Textarea.demo.js. You can start the docs locally by running 'yarn doc:start'

File: packages/Textarea/src/textarea.scss (hunk 1 above)
Comment (vicky-comeau) [reply packages/Textarea/src/textarea.scss:82]: It was there to prevent this overlap:
image

File: packages/Textarea/src/textarea.scss [hunk 3]
```diff
@@ -2,94 +2,145 @@
...
+        &::placeholder {
+            color: var(--ids-textarea-color-placeholder);
+        }
+
+        &:focus:not(&--error),
+        .ids-textarea--focus:not(&--error) & {
+            border-color: var(--ids-textarea-focus-border-color);
+        }
+
+        &:hover:not(:disabled) {
```
Comment (vicky-comeau) [inline packages/Textarea/src/textarea.scss:113]: I prefer not to use :not if it's not needed and can just be overridden normally. The reason being that it becomes more specific and can end up being more annoying later down the road when others use it.

**Comment by franckgaudin** [comment]
> I don't think this is normal...
> image
> It's on the docs so the file is Textarea.demo.js. You can start the docs locally by running 'yarn doc:start'
> It's also visible in storybook for Igloo.

It's not just in the doc, storybook also has this error 

textarea-rebrand

File: packages/Textarea/src/textarea.scss [hunk 4]
```diff
@@ -2,94 +2,145 @@
...
+    /* Error */
+    --ids-textarea-error-border-color: var(--hop-danger-border-strong);
+
+    /* Disabled */
+    --ids-textarea-disabled-background: var(--hop-neutral-border-disabled);
+    --ids-textarea-disabled-color: var(--hop-neutral-surface-disabled);
+
+    /* Character Indicator */
+    --ids-textarea-char-indicator-bottom: 0.5rem;
+    --ids-textarea-char-indicator-right: 1rem;
```
Comment (vicky-comeau) [inline packages/Textarea/src/textarea.scss:73]: bottom and right are similar to margin. I would use a token for this.

File: packages/Textarea/src/textarea.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/Textarea/src/textarea.scss:146]: ok igloo uses the border, but my first comment still stands :P

File: packages/Textarea/src/textarea.scss (hunk 4 above)
Comment (fraincs) [reply packages/Textarea/src/textarea.scss:73]: I agree if these emulate a margin I am ok with it.

File: packages/Textarea/src/textarea.scss (hunk 3 above)
Comment (fraincs) [reply packages/Textarea/src/textarea.scss:113]: I get where you are coming from. Not is my best friend especially for disabled state, I'll override the disabled hover, I tend to prefer to avoid adding unnecessary CSS but I get the why.

File: packages/Textarea/src/textarea.scss [hunk 5]
```diff
@@ -2,94 +2,156 @@
...
+        font-size: var(--ids-textarea-char-indicator-size);
+        line-height: var(--ids-textarea-char-indicator-line-height);
+        position: absolute;
+        right: var(--ids-textarea-char-indicator-right);
     }
-  }
-
-  &__char-indicator {
-    bottom: var(--ids-textarea-char-indicator-bottom);
-    color: var(--ids-textarea-char-indicator-color);
```
Comment (vicky-comeau) [inline packages/Textarea/src/textarea.scss:89]: Do we not need a color anymore? If this is not needed, please remove the variable at the top.

Comment (fraincs) [reply packages/Textarea/src/textarea.scss:89]: good catch I removed it by mistake.

**Review by vicky-comeau — DISMISSED** [review]

**Review by vicky-comeau — APPROVED** [review]

**Review by franckgaudin — APPROVED** [review]
